### PR TITLE
Add java_parameters kt_kotlinc_option

### PR DESCRIPTION
**Pull request by sq-msamuel**
The java_parameters flag allows Java 1.8 reflection to see the names of method parameters.

This is needed for some frameworks, for example Apache Beam schemas

### CODE REVIEW

File: kotlin/internal/opts.bzl [hunk 1]
```diff
@@ -86,6 +86,13 @@ _KOPTS = {
         ),
         type = attr.bool,
     ),
+    "java_parameters": struct(
+        args = dict(
+            default = False,
+            doc = "Generate metadata for Java 1.8 reflection on method parameters.",
```
Comment (cgruber) [inline kotlin/internal/opts.bzl:92]: Can you say "1.8+" since it exists since then?

**Review by cgruber — APPROVED** [review]

**Comment by sq-msamuel** [comment]
I don't know why CLA bot doesn't like that commit... it was authored by me :(

**Comment by cgruber** [comment]
I never understand CLAbot. We used to get along once.

**Comment by cgruber** [comment]
Wanna consider squashing into one commit, and see if it reacts better to that one?

**Comment by sq-msamuel** [comment]
Ok, I've made a mess of this one too, new PR incoming

**Pull request closed without merging**